netcfgbu/config_model.py

```diff
@@ -35,121 +39,210 @@
+        email: Optional email for Git operations.
+        username: Optional username for Git authentication.
+        password: Optional password for Git authentication.
+        token: Optional token for Git authentication.
+        deploy_key: Optional deploy key file path for Git authentication.
+        deploy_passphrase: Optional passphrase for the deploy key.
+    """
+
+    name: Optional[str] = None
     repo: EnvExpand
-    email: Optional[str]
-    username: Optional[EnvExpand]
-    password: Optional[EnvExpand]
-    token: Optional[EnvSecretStr]
-    deploy_key: Optional[FilePathEnvExpand]
-    deploy_passphrase: Optional[EnvSecretStr]
-
-    @validator("repo")
-    def validate_repo(cls, repo):  # noqa
+    add_tag: Optional[bool] = False
+    email: Optional[str] = None
+    username: Optional[EnvExpand] = None
+    password: Optional[EnvExpand] = None
+    token: Optional[EnvSecretStr] = None
+    deploy_key: Optional[FilePathEnvExpand] = None
+    deploy_passphrase: Optional[EnvSecretStr] = None
+
+    @field_validator("repo")
+    @classmethod
+    def validate_repo(cls, repo: str) -> str:
+        """Validate the Git repository URL.
+
+        Args:
+            repo: The repository URL.
+
+        Returns:
+            str: The validated repository URL.
+
+        Raises:
+            ValueError: If the repository URL does not start with the expected protocols.
+        """
         expected = ("https:", "git@")
         if not repo.startswith(expected):
-            raise ValueError(
-                f"Bad repo URL [{repo}]: expected to start with {expected}."
-            )
+            raise ValueError(f"Bad repo URL [{repo}]: expected to start with {expected}.")
         return repo
 
-    @root_validator
-    def enure_proper_auth(cls, values):
+    @model_validator(mode="before")
+    @classmethod
+    def ensure_proper_auth(cls, values: dict) -> dict:
+        """Ensure that the correct authentication method is provided.
+
+        Args:
+            values: The dictionary of values provided to the model.
+
+        Returns:
+            dict: The validated values.
+
+        Raises:
+            ValueError: If no or multiple authentication methods are provided, or
+            if deploy_passphrase is used without deploy_key.
+        """
         req = ("token", "deploy_key", "password")
         auth_vals = list(filter(None, (values.get(auth) for auth in req)))
-        auth_c = len(auth_vals)
-        if not auth_c:
-            raise ValueError(
-                f'Missing one of required auth method fields: {"|".join(req)}'
-            )
+        auth_count = len(auth_vals)
+        if auth_count == 0:
+            raise ValueError(f'Missing one of required auth method fields: {"|".join(req)}')
 
-        if auth_c > 1:
+        if auth_count > 1:
             raise ValueError(f'Only one of {"|".join(req)} allowed')
 
         if values.get("deploy_passphrase") and not values.get("deploy_key"):
```

Comment:
**Suggestion:** In the `ensure_proper_auth` method of the `GitSpec` class, the check for `deploy_passphrase` without `deploy_key` should be moved outside the loop that checks for multiple authentication methods to ensure it is always evaluated. [possible issue, importance: 6]
```suggestion
if values.get("deploy_passphrase") and not values.get("deploy_key"):
    raise ValueError("deploy_passphrase provided without deploy_key")
```